compact/hcal/forward_insert.xml

```diff
@@ -23,6 +23,24 @@
     <constant name="HcalEndcapPInsertESRFoilThickness"           value="0.015*cm"/>
 
     <constant name="LFHCAL_FrontWallThickness"                   value="1.50*cm"/>
+    <documentation>
+      #### Cell Sizes
+    </documentation>
+    <constant name="HcalEndcapPInsertCellSizeHG" value="2.16*cm"/>
+    <constant name="HcalEndcapPInsertCellSizeLGLeft" value="2.84*cm"/>
+    <constant name="HcalEndcapPInsertCellSizeLGRight" value="3.1*cm"/>
+    <documentation>
+      #### Transverse offsets of cells
+    </documentation>
+    <constant name="HcalEndcapPInsertCellToPCBEdge" value="0.04*cm"/>
+    <constant name="HcalEndcapPInsertOffsetXHGRight" value="-HcalEndcapPInsert_width/2.+HcalEndcapPInsertCellToPCBEdge-HcalEndcapPInsertCellSizeHG*0.25"/>
+    <constant name="HcalEndcapPInsertOffsetYHGRight" value="HcalEndcapPInsertCellSizeHG*sqrt(3)/2*0"/>
+    <constant name="HcalEndcapPInsertOffsetXHGLeft" value="HcalEndcapPInsert_width/2.-HcalEndcapPInsertCellToPCBEdge-HcalEndcapPInsertCellSizeHG*0"/>
+    <constant name="HcalEndcapPInsertOffsetYHGLeft" value="HcalEndcapPInsertCellSizeHG*sqrt(3)/2*0"/>
+    <constant name="HcalEndcapPInsertOffsetXLGRight" value="-HcalEndcapPInsert_width/2.+HcalEndcapPInsertCellToPCBEdge-HcalEndcapPInsertCellSizeLGRight*0.5"/>
+    <constant name="HcalEndcapPInsertOffsetYLGRight" value="HcalEndcapPInsertCellSizeLGRight*0*sqrt(3)/2"/>
+    <constant name="HcalEndcapPInsertOffsetXLGLeft" value="HcalEndcapPInsert_width/2.-HcalEndcapPInsertCellToPCBEdge+HcalEndcapPInsertCellSizeLGLeft*0.5"/>
+    <constant name="HcalEndcapPInsertOffsetYLGLeft" value="HcalEndcapPInsertCellSizeLGLeft*0*sqrt(3)/2"/>
```

Comment:
For the x offset, I first added or subtracted the ((PCB width)/2-margin).  This way the cells on the left part of the detector are determined relative to the left edge of the detector and those on the right are determined relative to the right part of the detector.  From there, it was trial and error with adding multiples of (cell side length)/4 for x and (cell side length)*sqrt(3)/2 to get the positions to match those that I had made with the algorithm.  Why multiples of these numbers?  Because the offsets in the staggered segmentation were multiples of these values, and so intuitively it made sense to do it like this instead of racking my brain to reverse engineer my earlier staggered segmentations code.